bofire/domain/constraint.py

```diff
@@ -38,6 +40,18 @@ def __call__(self, experiments: pd.DataFrame) -> pd.Series:
         """
         pass
 
+    @abstractmethod
```

Comment:
Do we want to enforce jacobian implementation for all constraints? This means for example also a Jacobian for NChooseK which is not really possible as it is in principal non-differentiable in its general form. Of course there are continuous relaxations for it, I would just implement it for those constraints where it is possible and not add it as abstract method for the base class.

Reply:
I agree, we do not need to have a jacobian implementation for every type of constraint. Removed the decorator.